Marlin/temperature.cpp

```diff
@@ -122,16 +132,33 @@ static volatile bool temp_meas_ready = false;
 
 #ifdef PIDTEMP
   //static cannot be external:
-  static float temp_iState[EXTRUDERS] = { 0 };
-  static float temp_dState[EXTRUDERS] = { 0 };
-  static float pTerm[EXTRUDERS];
-  static float iTerm[EXTRUDERS];
-  static float dTerm[EXTRUDERS];
-  //int output;
-  static float pid_error[EXTRUDERS];
-  static float temp_iState_min[EXTRUDERS];
-  static float temp_iState_max[EXTRUDERS];
-  static bool pid_reset[EXTRUDERS];
+  #ifndef SINGLENOZZLE
```

Comment:
Try this instead:

    #ifdef SINGLENOZZLE
      #define NOZZLES 1
    #else
      #define NOZZLES EXTRUDERS
    #endif
    static float temp_iState[NOZZLES] = { 0 };
    static float temp_dState[NOZZLES] = { 0 };
    static float pTerm[NOZZLES];
    static float iTerm[NOZZLES];
    static float dTerm[NOZZLES];
    //int output;
    static float pid_error[NOZZLES];
    static float temp_iState_min[NOZZLES];
    static float temp_iState_max[NOZZLES];
    // static float pid_input[NOZZLES];
    // static float pid_output[NOZZLES];
    static bool pid_reset[NOZZLES];